samples/deployment/aws/build_ami_vftd.yaml

```
@@ -0,0 +1,196 @@
+            cidr_ip: 0.0.0.0/0
+      register: stage_vpc_security_group
+
+    # Create VM Section
+    - name: "Create staging VM"
+      ec2:
+        key_name: "{{ ansible_ssh_private_key_file }}"
+        instance_type: "{{ ec2_instance_type }}"
+        image: "{{ ec2_base_image }}"
+        group: "{{ stage_vpc_security_group.group_name }}"
+        vpc_subnet_id: "{{ stage_subnet.subnet.id }}"
+        volumes:
+          - device_name: '/dev/sda1'
+            volume_size: 40
+            delete_on_termination: true
+          - device_name: '/dev/sdc'
+            volume_size: 52
+            delete_on_termination: true
+        wait: yes
+        assign_public_ip: yes
+        instance_initiated_shutdown_behavior: terminate
+      register: stage_vm
+    - name: "Add the newly created host so that we can further contact it"
+      add_host:
+        name: "{{ item.public_ip }}"
+        groups: "stage_servers"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Wait for SSH to come up"
+      delegate_to: "{{ item.public_dns_name }}"
+      wait_for_connection:
+        delay: 60
+        timeout: 320
+      with_items: "{{ stage_vm.instances }}"
+
+    # Build vFTD Volume Section
+    - name: "Prepare mnt directory"
+      shell: "chmod 777 /mnt"
+      become: yes
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Copy QCOW2 file to the remote VM"
+      copy:
+        src: "./Cisco_Firepower_Threat_Defense_Virtual-{{ ftd_build_id }}.qcow2"
+        dest: "/mnt/image.qcow2"
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Copy fixup.sh file to the remote VM"
+      copy:
+        src: "./fixup-ftd.sh"
+        dest: "/mnt/fixup.sh"
+        mode: "u+rwx"
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Run fixup.sh preinstall command"
+      shell: "/mnt/fixup.sh preinstall"
+      become: yes
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Run fixup.sh doinstall command"
+      shell: "/mnt/fixup.sh doinstall"
+      become: yes
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Run fixup.sh postinstall command"
+      shell: "/mnt/fixup.sh postinstall"
+      become: yes
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+
+    # Create Custom AMI Section
+    - name: "Get volume ID"
+      ec2_instance_facts:
+        instance_ids:
+          - "{{ stage_vm.instance_ids.0 }}"
+      register: stage_vm_facts
+    - name: "Create a Snapshot"
+      ec2_snapshot:
+        instance_id: "{{ stage_vm.instance_ids.0 }}"
+        device_name: "/dev/sdc"
+        description: "Snapshot for vFTD AMI image"
+      register: stage_snapshot
+    - name: "Create Snapshot based AMI Image"
```

```suggestion
    - name: "Create a Snapshot-based AMI file"
```
I guess, I in AMI already stands for an Image.

---

yep. fixed